src/model/Series.ts

```diff
@@ -569,16 +569,22 @@ class SeriesModel<Opt extends SeriesOption = SeriesOption> extends ComponentMode
         }
 
         const data = this.getData(dataType);
-        if (data.getItemModel<StatesOptionMixin<unknown, unknown>>(dataIndex).get(['select', 'disabled'])) {
-            return false;
-        }
-
+        
         if (selectedMap === 'all') {
-            return true;
+            return !data.getItemModel<StatesOptionMixin<unknown, unknown>>(dataIndex).get(['select', 'disabled']);
         }
 
         const nameOrId = getSelectionKey(data, dataIndex);
-        return selectedMap[nameOrId] || false;
+
+        if (selectedMap[nameOrId] != null) {
```

Comment:
`selectedMap[nameOrId] != null` is a redundant check. The whole logic can be simplified to:

```ts
return (selectedMap === 'all' || selectedMap[getSelectionKey(data, dataIndex)])
  && !data.getItemModel<StatesOptionMixin<unknown, unknown>>(dataIndex).get(['select', 'disabled'])
```

Whener you see a long line like `data.getItemModel<StatesOptionMixin<unknown, unknown>>(dataIndex).get(['select', 'disabled'])` multiple times in one function. Try to reduce it to one.